src/Compilers/CSharp/Portable/Binder/Binder_Lookup.cs

```diff
@@ -924,9 +924,52 @@ private static Symbol GetNearestOtherSymbol(ConsList<TypeSymbol> list, TypeSymbo
             Debug.Assert(type.IsInterface);
 
             LookupMembersWithoutInheritance(current, type, name, arity, options, originalBinder, accessThroughType, diagnose, ref useSiteDiagnostics, basesBeingResolved);
-            if ((options & (LookupOptions.NamespaceAliasesOnly | LookupOptions.NamespacesOrTypesOnly)) == 0 && !originalBinder.InCrefButNotParameterOrReturnType)
+            if ((options & LookupOptions.NamespaceAliasesOnly) == 0 && !originalBinder.InCrefButNotParameterOrReturnType)
             {
-                LookupMembersInInterfacesWithoutInheritance(current, type.AllInterfacesWithDefinitionUseSiteDiagnostics(ref useSiteDiagnostics), name, arity, basesBeingResolved, options, originalBinder, accessThroughType, diagnose, ref useSiteDiagnostics);
+                ImmutableArray<NamedTypeSymbol> baseInterfaces;
+
+                if (basesBeingResolved?.Any() != true)
+                {
+                    baseInterfaces = type.AllInterfacesWithDefinitionUseSiteDiagnostics(ref useSiteDiagnostics);
+                }
+                else
+                {
+                    var interfacesLookedAt = PooledHashSet<NamedTypeSymbol>.GetInstance();
+                    var baseInterfacesBuilder = ArrayBuilder<NamedTypeSymbol>.GetInstance();
+
+                    getBaseInterfaces(type, baseInterfacesBuilder, interfacesLookedAt, basesBeingResolved);
+
+                    for (int i = 0; i < baseInterfacesBuilder.Count; i++)
+                    {
+                        getBaseInterfaces(baseInterfacesBuilder[i], baseInterfacesBuilder, interfacesLookedAt, basesBeingResolved);
+                    }
+
+                    interfacesLookedAt.Free();
+                    baseInterfaces = baseInterfacesBuilder.ToImmutableAndFree();
+
+                    foreach (var candidate in baseInterfaces)
+                    {
+                        candidate.OriginalDefinition.AddUseSiteDiagnostics(ref useSiteDiagnostics);
```

Comment:
>candidate.OriginalDefinition.AddUseSiteDiagnostics(ref useSiteDiagnostics); [](start = 24, length = 75)

I didn't understand this. Why are we collecting use site diagnostics from a base interface which may not end up being used/useful?

Reply:
> I didn't understand this. Why are we collecting use site diagnostics from a base interface which may not end up being used/useful?

1. This code is a replacement for AllInterfacesWithDefinitionUseSiteDiagnostics call.
2. In lookup we depend on all base interfaces, all of them are used.


---
In reply to: [322412154](https://github.com/dotnet/roslyn/pull/38564#discussion_r322412154) [](ancestors = 322412154)